In [4]:
%pip install kagglehub
%pip install pandas

Note: you may need to restart the kernel to use updated packages.
  Using cached pandas-2.2.3-cp312-cp312-macosx_11_0_arm64.whl.metadata (89 kB)
  Using cached pytz-2024.2-py2.py3-none-any.whl.metadata (22 kB)
  Using cached tzdata-2024.2-py2.py3-none-any.whl.metadata (1.4 kB)
Using cached pandas-2.2.3-cp312-cp312-macosx_11_0_arm64.whl (11.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.1/5.1 MB 476.1 kB/s eta 0:00:0000:0100:01
Using cached pytz-2024.2-py2.py3-none-any.whl (508 kB)
Using cached tzdata-2024.2-py2.py3-none-any.whl (346 kB)
Note: you may need to restart the kernel to use updated packages.


In [6]:
import kagglehub

# Téléchargement du dataset avec kagglehub
path = kagglehub.dataset_download("prathamsaraf1389/spotify-global-top-50-daily-update")

print("SPOTIFYYYYYY :", path)

# Remplace "playlist.csv" par le nom exact trouvé
file_path = os.path.join(path, "playlist.csv")
spotify_data = pd.read_csv(file_path)

# Affiche un aperçu des données
print(spotify_data.head())


SPOTIFYYYYYY : /Users/jbb/.cache/kagglehub/datasets/prathamsaraf1389/spotify-global-top-50-daily-update/versions/315
           track_name track_add_date track_add_time  multiple_artists_bool  \
0             Flowers     2023-03-31       10:02:16                  False   
1           Kill Bill     2023-03-31       10:02:16                  False   
2                 TQG     2023-03-31       10:02:16                   True   
3                BESO     2023-03-31       10:02:16                   True   
4  Boy's a liar Pt. 2     2023-03-31       10:02:16                   True   

                   name_of_artists               album_name  \
0                  ['Miley Cyrus']  Endless Summer Vacation   
1                          ['SZA']                      SOS   
2           ['KAROL G', 'Shakira']       MAÑANA SERÁ BONITO   
3    ['ROSALÍA', 'Rauw Alejandro']                       RR   
4  ['PinkPantheress', 'Ice Spice']       Boy's a liar Pt. 2   

  album_release_date album_release_

# Introduction

Fondée en 2006 par Daniel Ek & Martin Lorentzon en Suède, Spotify est devenue l'une des plateformes de streaming musical les plus populaires au monde, révolutionnant la manière d'écouter la musique en offrant un accès instantané à un vaste cataloque de morceaux, albums et podcast.

La plateforme propose une multitude de playlists dont la célèbre "Top 50 Global", une sélection des morceaux les plus écoutés dans le monde entier, cette playlist, étant mise à jour quotidiennement, reflète les goûts musicaux globaux en temps réel en se basant sur le nombre de streams, dont artistes, producteurs et labels, voient une vitrine essentielle pour mesure le succès de leurs titre et toucher une audience mondiale.

# Présentation de la data

Nous avons les lignes suivantes dans notre database

track_name: Le nom du morceau

track_add_date: La date de l'ajout du morceau dans la playlist

track_add_time: L'heure de l'ajout du morceau dans la playlist

multiple_artists_bool: Un booleen qui nous renseigne si le morceau est mono ou multi artiste

name_of_artists: Nom des artistes

album_name: Nom de l'album

album_release_date: Date de sortie de l'album

album_release_date_precision: Précision de la date de sortie de l'album

number_of_tracks_in_album: Nombre de morceaux dans l'album

position_in_playlist: Position du morceau de la playlist

track_duration_ms: Durée du morceau en milisecondes

track_popularity: Popularité du morceau 

track_explicit: Un booleen qui nous donne si le morceau est explicite ou non

images_path: Lien pour image du morceau


# Premières observations

In [11]:
spotify_data.describe()

number_of_tracks_in_album  position_in_playlist  track_duration_ms  \
count               15550.000000          15550.000000       15550.000000   
mean                   10.519807             25.500000      196956.244437   
std                     9.213601             14.431334       38362.990747   
min                     1.000000              1.000000       36226.000000   
25%                     1.000000             13.000000      172797.000000   
50%                    12.000000             25.500000      195986.000000   
75%                    17.000000             38.000000      222461.000000   
max                   119.000000             50.000000      405906.000000   

       track_popularity  
count      15550.000000  
mean          89.535691  
std           11.933126  
min            0.000000  
25%           88.000000  
50%           92.000000  
75%           95.000000  
max          100.000000

Nous avons 15550 morceaux dans ce dataset, cela correspond à 15550/50 = 311 semaines de Globat Top 50

On va pouvoir observer quels sont les morceaux les plus récurrents, le nombre d'artistes différents et la distribution de la durée des morceaux.